src/train_model.py

```
@@ -358,12 +514,153 @@ def train_yolov8(
+        device: Device to use for training (e.g., "0", "cpu", "cuda:0").
+            If None, automatically detects GPU availability.
+
+    Returns:
+        Dict containing training results and metrics.
+
+    Raises:
+        TrainingError: If training fails.
+    """
+    # Determine device to use for training
+    if device is None:
+        import torch  # pylint: disable=import-outside-toplevel
```

Pylint (4.0.4)

[error] Unable to import 'torch'

(E0401)